src/consensus.rs

```diff
@@ -320,24 +320,21 @@ impl <L, M> Consensus<L, M> where L: Log, M: StateMachine {
                             messages::append_entries_response_inconsistent_prev_entry(self.current_term(),
                                 leader_prev_log_index)
                         } else {
-                            let entries = request.get_entries().unwrap();
-                            let num_entries = entries.len();
-                            scoped_debug!("AppendEntriesRequest: {} entries from leader: {}",
-                                          num_entries, from);
-                            if num_entries > 0 {
-                                let mut entries_vec = Vec::with_capacity(num_entries as usize);
-                                for i in 0..num_entries {
-                                    let entry = entries.get(i);
-                                    let term = Term::from(entry.get_term());
-                                    let data = entry.get_data().unwrap();
-                                    entries_vec.push((term, data));
-                                }
+                            if let Ok(entries) = request.get_entries() {
```

Comment:
If I'm reading this correctly, the error handling is being changed to silently ignore failures to read the entries from the message, or the contents of entries.  If that's so, I think it's still better to keep the `unwrap` and crash on failure. We definitely need better error handling in this section (and all of `consensus.rs`, but I'm not sure the answer is to silently drop data on deserialization failure and return success.  We should come up with a plan on how to handle errors stemming from cap'n proto, and the persistent log in `consensus.rs`, but for now I don't think crashing is a bad solution.